swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/BillableUsageAggregateConsumer.java

```diff
@@ -169,7 +169,8 @@ private boolean isUsageDateValid(BillableUsageAggregate aggregate) {
     OffsetDateTime startOfCurrentHour =
         OffsetDateTime.now(Clock.systemUTC()).truncatedTo(ChronoUnit.HOURS);
     OffsetDateTime cutoff = startOfCurrentHour.minus(azureUsageWindow);
-    return !aggregate.getWindowTimestamp().isBefore(cutoff);
+    var earliestUsageDate = aggregate.getSnapshotDates().stream().sorted().findFirst();
+    return !earliestUsageDate.map(date -> date.isBefore(cutoff)).orElse(false);
```

Comment:
what is the motivation of this change? I don't see any info in the PR description or in the JIRA ticket.

Reply:
I realized while testing we have an issue here, we need to use snapshot dates and not the aggregate window for this. It belongs in a new card though I'll remove from this PR and make a new card for it.